common/memcached/templates/deployment.yaml

```
@@ -93,4 +93,6 @@ spec:
             memory: {{ .Values.metrics.resources.requests.memory | quote }}
         {{- end }}
       {{- end }}
+      {{- if not (and (hasKey $.Values "priorityClass") (hasKey $.Values.priorityClass "disabled") ($.Values.priorityClass.disabled)) }}
       priorityClassName: {{ default "openstack-service-critical" .Values.priority_class | quote }}
+      {{- end }}
```

This interface looks unnecessarily confusing. Why have `.Values.priority_class` be a string and then use the different key `.Values.priorityClass` to hold `{"disabled": true}`? I would suggest using a sentinel value:

```suggestion
      {{- $priority_class := .Values.priority_class | default "openstack-service-critical" }}
      {{- if ne $priority_class "none" }}
      priorityClassName: {{ quote $priority_class }}
      {{- end }}
```

Or even simpler, declare `priority_class: openstack-service-critical` in this chart's values.yaml and allow using charts to disable it by overriding that to null. Then you only need:

```suggestion
      {{- if .Values.priority_class }}
      priorityClassName: {{ quote .Values.priority_class }}
      {{- end }}
```